src/raft_log.c

```diff
@@ -241,14 +241,14 @@ int log_delete(log_t* me_, int idx)
     log_private_t* me = (log_private_t*)me_;
     int e = 0;
 
-    if (0 == me->count || !has_idx(me, idx))
+    if (!has_idx(me, idx))
         return -1;
 
-    while (me->base + 1 + me->count > idx)
+    while (idx <= me->base + me->count)
     {
-        raft_entry_t *ptr = &me->entries[me->back - 1];
-        int k = batch_down(me, me->base + 1 + me->count - 1,
-                               me->base + 1 + me->count - idx);
+        int k = batch_down(me, me->base + me->count,
+                               me->base + me->count - idx + 1);
+        raft_entry_t *ptr = &me->entries[me->back - 1 - k];
```

Comment:
Hmm, should we write "[me->back - k]" instead of "[me->back - 1 - k]"?

Actually, is subtracting from me->back safe, or not... For instance, if me->count = me->size and me->front = 0, then me->back = 0? Not a problem of this pull request, but perhaps we should use "subscript(me->base + me->count - k + 1)" instead. And pehraps eventually remove the redundant me->back field.

Reply:
You're right. I have removed the back field completely in the latest commit.